sql/src/main/java/io/druid/sql/calcite/rel/DruidQuery.java

```diff
@@ -540,18 +608,26 @@ public VirtualColumns getVirtualColumns(final ExprMacroTable macroTable, final b
   {
     final List<VirtualColumn> retVal = new ArrayList<>();
 
-    if (grouping != null) {
-      if (includeDimensions) {
-        for (DimensionExpression dimensionExpression : grouping.getDimensions()) {
-          retVal.addAll(dimensionExpression.getVirtualColumns(macroTable));
+    if (selectProjection != null) {
+      retVal.addAll(selectProjection.getVirtualColumns());
+    } else {
+      if (sortProject != null) {
+        for (Aggregation aggregation : sortProject.getPostAggregators()) {
+          retVal.addAll(aggregation.getVirtualColumns());
```

Comment:
There will never be any virtual columns added by post-aggregators (virtual columns can only be added by aggregators that read the input data).

This code would be removed naturally if `getPostAggregators` was changed to return `List<PostAggregator>` rather than `List<Aggregation>`, so that's a point in favor of changing those types.

Reply:
Removed.